pkg/vpcmodel/nodesExplainability.go

```diff
@@ -30,16 +30,48 @@ type srcDstDetails struct {
 
 type rulesAndConnDetails []*srcDstDetails
 
-type explanation struct {
+type ExplanationArgs struct {
+	src        string
+	dst        string
+	protocol   string
+	srcMinPort int64
+	srcMaxPort int64
+	dstMinPort int64
+	dstMaxPort int64
+}
+
+func NewExplanationArgs(src, dst, protocol string, srcMinPort, srcMaxPort, dstMinPort, dstMaxPort int64) *ExplanationArgs {
+	return &ExplanationArgs{src: src, dst: dst, protocol: protocol,
+		srcMinPort: srcMinPort, srcMaxPort: srcMaxPort, dstMinPort: dstMinPort, dstMaxPort: dstMaxPort}
+}
+
+type Explanation struct {
 	c               *VPCConfig
 	connQuery       *common.ConnectionSet
 	rulesAndDetails *rulesAndConnDetails // rules and more details for a single src->dst
+	src             string
+	dst             string
 	// grouped connectivity result:
 	// grouping common explanation lines with common src/dst (internal node) and different dst/src (external node)
 	// [required due to computation with disjoint ip-blocks]
 	groupedLines []*groupedConnLine
 }
 
+func translateCDtoConnectionSet(protocol string, srcMinPort, srcMaxPort, dstMinPort, dstMaxPort int64) *common.ConnectionSet {
```

Comment:
- this function can have one input arg of type `ExplanationArgs` instead
- rename function to follow go naming conventions